mParticleCore.brs

```diff
@@ -138,6 +138,94 @@ function mParticleConstants() as object
                                 end function
         
     }
+    '
+    ' Consent Management
+    '
+    ConsentState = {
+        build : function ()
+            consentState = {}
+            consentState.gdpr = {}
+            consentState.ccpa = {}
+            return consentState
+        end function,
+
+        ' GDPR
+        addGDPRConsentState : function (consentState as object, purpose as string, gdprConsent as object)
+            consentState.gdpr.AddReplace(purpose, gdprConsent)
+        end function,
+        getCCPAConsentState : function(consentState as object)
+            return consentState.ccpa
+        end function,
+        getGDPRConsentState : function(consentState as object)
+            return consentState.gdpr
+        end function,
+        removeGDPRConsentState : function (consentState as object, purpose as string)
+            if (consentState.gdpr.DoesExist(purpose)) then
+                consentState.gdpr.Delete(purpose)
+            end if
+        end function,
+
+        ' CCPA
+        setCCPAConsentState : function (consentState as object, ccpaConsent as object)
+            consentState.ccpa.AddReplace("data_sale_opt_out", ccpaConsent)
+        end function,
+        removeCCPAConsentState : function (consentState as object)
+            consentState.ccpa.Delete("data_sale_opt_out")
+        end function
+    }
+    CCPAConsentState = {
+        build : function (document="" as string, consented=false as boolean, location="" as string, timestamp=0 as longInteger, hardwareId="" as string)
+            ccpaConsentState = {}
+            ccpaConsentState.d = document
+            ccpaConsentState.c = consented
+            ccpaConsentState.l = location
+            ccpaConsentState.ts = timestamp
+            ccpaConsentState.h = hardwareId
+            return ccpaConsentState
+        end function
+        setDocument : function (ccpaConsentState as object, document as string)
+            ccpaConsentState.d = document
+        end function,
+        setConsented : function (ccpaConsentState as object, consented as boolean)
+            ccpaConsentState.c = consented
+        end function,
+        setLocation : function (ccpaConsentState as object, location as string)
+            ccpaConsentState.l = location
+        end function,
+        setTimestamp : function (ccpaConsentState as object, timestamp as LongInteger)
+            ccpaConsentState.ts = timestamp
+        end function,
+        setHardwareId : function (ccpaConsentState as object, hardwareId as string)
+            ccpaConsentState.h = hardwareId
+        end function,
+    }
+    GDPRConsentState = {
+        build : function (document="" as string, consented=false as boolean, location="" as string, timestamp=0 as longInteger, hardwareId="" as string)
```

Comment:
is there any reason to default these to empty string? better to upload them as null if not included. same comment above on line 177.